pageserver/src/walredo.rs

```diff
@@ -202,9 +202,9 @@ impl PostgresRedoManager {
     /// rely on our owner calling this function periodically in its own housekeeping
     /// loops.
     pub(crate) fn maybe_quiesce(&self, idle_timeout: Duration) {
-        if let Ok(g) = self.last_redo_at.try_lock() {
-            if let Some(last_redo_at) = *g {
-                if last_redo_at.elapsed() >= idle_timeout {
+        if let Ok(g) = self.last_successful_redo_at.try_lock() {
```

Comment:
It would be good to have a comment here explaining why we will kill walredo even if it's busy, if we haven't seen success for some time.  It's a valid behavior but I think it's a bit nuanced and the reader might wonder "if we kill failing walredo processes, why wait rather than kill them immediately" etc.  Basically some of the PR description text should go in the code as a comment :-)

Reply:
https://github.com/neondatabase/neon/pull/6568/commits/91cec9ba48953bc050a6bc8d3914c9416b5ce035